test/v200/utils/api/test_utils.go

```diff
@@ -0,0 +1,173 @@
+	"strconv"
+	"strings"
+	"testing"
+
+	common "github.com/devfile/api/v2/test/v200/utils/common"
+	"github.com/santhosh-tekuri/jsonschema"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// numDevfiles : the number of devfiles to create for each test
+	numDevfiles = 5
+
+	schemaFileName = "../../../schemas/latest/ide-targeted/devfile.json"
+)
+
+var schemas = make(map[string]SchemaFile)
+
+type SchemaFile struct {
+	Schema *jsonschema.Schema
+}
+
+type DevfileValidator struct{}
+
+// WriteAndVerify implements Saved.DevfileValidator interface.
+// writes to disk and validates the specified devfile
+func (devfileValidator DevfileValidator) WriteAndValidate(devfile *common.TestDevfile) error {
+	err := writeDevfile(devfile)
+	if err != nil {
+		common.LogErrorMessage(fmt.Sprintf("Error writing file : %s : %v", devfile.FileName, err))
+	} else {
+		err = validateDevfile(devfile)
+		if err != nil {
+			common.LogErrorMessage(fmt.Sprintf("Error vaidating file : %s : %v", devfile.FileName, err))
+		}
+	}
+	return err
+}
+
+// CheckWithSchema checks the validity of aa devfile against the schema.
+func (schemaFile *SchemaFile) CheckWithSchema(devfile string, expectedMessage string) error {
+
+	// Read the created yaml file, ready for converison to json
+	devfileData, err := ioutil.ReadFile(devfile)
+	if err != nil {
+		common.LogErrorMessage(fmt.Sprintf("  FAIL: schema : unable to read %s: %v", devfile, err))
+		return err
+	}
+
+	// Convert the yaml file to json
+	devfileDataAsJSON, err := yaml.YAMLToJSON(devfileData)
+	if err != nil {
+		common.LogErrorMessage(fmt.Sprintf("  FAIL : %s : schema : failed to convert to json : %v", devfile, err))
+		return err
+	}
+
+	validationErr := schemaFile.Schema.Validate(bytes.NewReader(devfileDataAsJSON))
+	if validationErr != nil {
+		if len(expectedMessage) > 0 {
+			if !strings.Contains(validationErr.Error(), expectedMessage) {
+				err = errors.New(common.LogErrorMessage(fmt.Sprintf("  FAIL : schema : %s : Did not fail as expected : %s  got : %v", devfile, expectedMessage, validationErr)))
+			} else {
+				common.LogInfoMessage(fmt.Sprintf("PASS: schema :  Expected Error received : %s", expectedMessage))
+			}
+		} else {
+			err = errors.New(common.LogErrorMessage(fmt.Sprintf("  FAIL : schema : %s : Did not pass as expected, got : %v", devfile, validationErr)))
+		}
+	} else {
+		if len(expectedMessage) > 0 {
+			err = errors.New(common.LogErrorMessage(fmt.Sprintf("  FAIL : schema : %s :  was valid - Expected Error not found : %v", devfile, validationErr)))
+		} else {
+			common.LogInfoMessage(fmt.Sprintf("  PASS : schema : %s : devfile was valid.", devfile))
+		}
+	}
+	return err
+}
+
+// GetSchema downloads and saves a schema from the provided url
+func GetSchema(schemafile string) (SchemaFile, error) {
+
+	var err error
+	schemaFile, found := schemas[schemafile]
+	if !found {
+
+		schemaFile = SchemaFile{}
+
+		// Prepare the schema file
+		compiler := jsonschema.NewCompiler()
+		compiler.Draft = jsonschema.Draft7
```

Comment:
To be honest I have no idea what draft 7 is, the tool I use offers Drafts 4, 6 and 7. Using 7 because it was the biggest number and it works. Is there a specific one I should use?

Reply:
Added a comment to that effect